haskala/modules/custom/haskala_book/haskala_book.module

```diff
@@ -16,6 +16,56 @@ function haskala_book_ctools_plugin_directory($module, $plugin) {
 }
 
 /**
+ *  Translation tab - Other names for the original author.
+ *
+ *  @param $person_nid
+ *   The author person node ID.
+ *
+ *  @return  The created field for orginal author other names.
+ */
+function haskala_book_author_names($person_nid) {
+  $query = new EntityFieldQuery();
+  $query->entityCondition('entity_type', 'node')
+    ->entityCondition('bundle', 'person')
+    ->propertyCondition('status', NODE_PUBLISHED)
+    ->propertyCondition('nid', $person_nid);
+  $result = $query->execute();
+  if (empty($result['node'])) {
+    return;
+  }
+  $author_node = node_load(reset($result['node'])->nid);
+  $author_wrapper = entity_metadata_wrapper('node', $author_node);
+
+  $author_names = array();
+
+  // Try build field_view_field().
+  if ($author_wrapper->field_hebrew_name) {
+    $author_names_as_field = field_view_field('node', $author_node, 'field_hebrew_name');
+    $author_names[] = $author_wrapper->field_hebrew_name->value();
+  }
+  if ($author_wrapper->field_german_name) {
+    if (!isset($author_names_as_field)) {
+      $author_names_as_field = field_view_field('node', $author_node, 'field_german_name');
```

Comment:
Probably we don't need to use field_view_field three times. Better would be to return the name as a string, and handle the label elsewhere